modules/single_page_checkout/EED_Single_Page_Checkout.module.php

```diff
@@ -746,18 +741,16 @@ private function _load_and_instantiate_reg_steps()
                     return false;
                 }
             }
-            EE_Registry::instance()->CFG->registration->skip_reg_confirmation = true;
-            EE_Registry::instance()->CFG->registration->reg_confirmation_last = true;
-            // skip the registration_confirmation page ?
-            if (EE_Registry::instance()->CFG->registration->skip_reg_confirmation) {
-                // just remove it from the reg steps array
-                $this->checkout->remove_reg_step('registration_confirmation', false);
-            } else if (
-                isset($this->checkout->reg_steps['registration_confirmation'])
-                && EE_Registry::instance()->CFG->registration->reg_confirmation_last
-            ) {
-                // set the order to something big like 100
-                $this->checkout->set_reg_step_order('registration_confirmation', 100);
+            if(isset($this->checkout->reg_steps['registration_confirmation'])){
+                // skip the registration_confirmation page ?
+                if (EE_Registry::instance()->CFG->registration->skip_reg_confirmation) {
+                    // just remove it from the reg steps array
+                    $this->checkout->remove_reg_step('registration_confirmation', false);
+                } elseif (EE_Registry::instance()->CFG->registration->reg_confirmation_last
+                ) {
+                    // set the order to something big like 100
+                    $this->checkout->set_reg_step_order('registration_confirmation', 100);
+                }
```

Comment:
I'm guessing the changes made here are why the boolean was changed in my earlier comment?